### ISSUE TITLE
[Feature Request] Auto-update preview

### ISSUE THREAD

**Issue by McGeifer** [Frederisk/Wikitext-VSCode-Extension#26]
I would love to see some kind of auto-update functionality for the preview window.

Right now you need to close and re-open the window every time you change something on your document. Especially when you playing with the visual style of you document it becomes a bit cumbersome.

**Comment by Frederisk** [comment]
Re-open? No no, just click preview button or execute the command again. The content of WebView will be refreshed.

**Comment by McGeifer** [comment]
I may have expressed myself a little unclearly. I use VS Code with two windows side by side, the wiki "code" on the left and the preview window on the right. This means that the preview button is no longer visible. Using Ctrl + Shift + V will refresh the preview page but will also jump to the beginning of the page so you always need to scroll down to the place where you did you changes. I was wondering if this could be improved.

**Comment by Frederisk** [comment]
You mean you open two windows at the same time instead of two tabs? If there are two tab pages, if the active tab page is source code editing, the button will appear again.

It may be difficult to automatically jump to the active editing position. The conversion from Wikitext to HTML relies on external APIs from your host, and I have no way to get a handle through the API to locate the context.

For an alternative method, you can open another tab and only preview that part.

**Comment by McGeifer** [comment]
With two open tabs, clicking the button will refresh the page 👍
But would it be possible to auto update the preview every x seconds or "simply" when typing in new stuff?

**Comment by Frederisk** [comment]
In fact, I considered the automatic refresh a long time ago. But as you just said, Refreshing the page will cause the page to return to the top. It means that if is regularly executed, the user maybe repeatedly interrupted by this action.

On the other hand, if it's refreshed with user's text input, then the access to the APIs may become very frequent, so that the speed limit is reached shortly.

In summary, There is no good idea to achieve this function currently.

**Comment by McGeifer** [comment]
Sounds reasonable, thanks for the information (and the nice tool of course).

**Comment by Frederisk** [comment]
This issue will remain open, to provide references for others. But this feature may not be added in the near future.
